tests/e2e/support/page/publicLinkPage.ts

```diff
@@ -0,0 +1,68 @@
+import { errors, Page } from 'playwright'
+import util from 'util'
+import { state } from '../../cucumber/environment/shared'
+import { PublicLink } from './files/publicLink'
+
+export class PublicLinkPage {
+  private static page: Page
+  private readonly passwordInput: string
+  private readonly passwordSubmitButton: string
+  private readonly fileUploadInput: string
+  private publicResourceSelector: string
+  private uploadedResourceSelector: string
+
+  constructor() {
+    this.passwordInput = 'input[type="password"]'
```

Comment:
didn't you guys use this type of locator assignment? :thinking: 
```suggestion
    this.passwordInput = SomePage.page.locator('input[type="password"]')
```
will be helpful for later usage.

Reply:
@kiranparajuli589  can you please look at line number 35. I think what you are suggesting me is that.

Reply:
we can't use `page.locator` there because the `page` doesn't exist when the constructor is called. There is a different function `setUp` that creates a page when needed and there is no actor for this page as it's unauthorize